### DISCUSSION TITLE
Add HasStagedChanges to the Branch model

### DISCUSSION THREAD

**Discussion by danielo515** [jesseduffield/lazygit#2728] [Ideas]
Related to #2724 
I think it will be cool if the current branch has a property indicating if it contains staged changes. This way you can construct better conditional flows based on the global state of the current branch.

**Comment by mark2185** [comment]
Having staged changes isn't really a property of a branch, in the `git` sense.

Is this regarding this specifically?

> Works like a charm, but it forces me to work using SelectedFile, while what I wanted in reality was current branch, but the current branch does not have the HasStagedChanges field

I mean, you don't have to use what `lazygit` has to offer, you can just invoke a `bash` script that checks if anything is staged and does whatever you want it to do.

**Reply by danielo515** [reply]
Ok, that could be a viable workaround. But what do you mean that staged changes is not a property of a branch?

**Reply by mark2185** [reply]
In my mental model (which may be wrong), branches are just a list of commits, they shouldn't have any information about the staging area.

Given that you can even stage things when you're in a `detached HEAD` state, a `HasStagedChanges` property would be more related to a _repository_, rather than _the current branch_.

**Reply by danielo515** [reply]
Yes, that makes sense. Is there any entity for the whole repository?

**Reply by mark2185** [reply]
Based on the current [models](https://github.com/jesseduffield/lazygit/tree/master/pkg/commands/models) there doesn't seem to be a way to do what you want.

I'm not sure how needed it is, maybe someone else will also want it. In the meantime, I'd advise a shell script.
